res/ps_border.fs.glsl

```diff
@@ -56,6 +56,57 @@ vec4 draw_dotted_edge() {
   return mix(white, circleColor, circleColor.a);
 }
 
+vec4 draw_double_edge(float pos, float len) {
+  // Devided border to 3 parts, draw color on first and third part,
+  // leave second part blank.
+  float one_third_len = len / 3.0;
+
+  float in_first_part = step(pos, one_third_len);
+  float in_third_part = step(len - one_third_len, pos);
+
+  // The result of this should be 1.0 if we're in the 1st or 3rd part.
+  // And 0.0 for the blank part.
+  float should_fill = in_first_part + in_third_part;
+
+  float color_weight = step(0.0, vF);
+  vec4 color = mix(vHorizontalColor, vVerticalColor, color_weight);
+
+  vec4 white = vec4(1.0, 1.0, 1.0, 1.0);
+  return mix(white, color, should_fill);
+}
+
+vec4 draw_double_edge_vertical() {
+  // Get our position within this specific segment
+  float position = vLocalPos.x - vLocalBorders.x;
+  return draw_double_edge(position, vLocalBorders.z);
+}
+
+vec4 draw_double_edge_horizontal() {
+  // Get our position within this specific segment
+  float position = vLocalPos.y - vLocalBorders.y;
+  return draw_double_edge(position, vLocalBorders.w);
+}
+
+vec4 draw_double_edge_with_radius() {
+  // Get our position within this specific segment
+  float position = distance(vRefPoint, vLocalPos) - vRadii.z;
+  float len = vRadii.x - vRadii.z;
+  return draw_double_edge(position, len);
+}
+
+vec4 draw_double_edge_corner() {
+  if (vRadii.x > 0) {
+    return draw_double_edge_with_radius();
+  }
+
+  bool is_vertical = (vBorderPart == PST_TOP_LEFT) ? vF < 0 : vF >= 0;
```

Comment:
The vF calculation is quiet confuse now. For the vertical case, only the vF on top left corner is lesser than 0. All other 3 corners are greater than 0. I just tweak the vF calculation to make it more consistent. After my tweak, the vF stand for vertical edge if vF > 0 and for horizontal edge if vF < 0 for all corners, without exception. So this `is_vertical` check would be more clear.